tools/detect_unused_modules

```diff
@@ -36,10 +36,12 @@ while [ "$#" -gt 0 ]; do
 done
 
 for test in $MODULES ; do
+    file="`git ls-files *$test*`"
```

Comment:
Why is this here?

Reply:
Sorry, fixed (I noticed yesterday, but forget to push).

Reply:
I also wonder whether this filter you tried to add in 407cfa0d9 shouldn't be changed in Makefile. It already filters tests:

```
test-unused-modules-changed:                                            
    @echo "[make] Unused modules check called over modified/new files only. For a full run use make test-unused-modules-full"
    tools/detect_unused_modules -m `git --no-pager diff --name-only --diff-filter=d origin/master | grep '^tests/*' | grep -v '^tests/test_pods/'`
    tools/detect_unused_modules -m `git --no-pager diff --unified=0 origin/master products/* | grep -oP "^-.*loadtest.*[\"']\K[^\"'].+(?=[\"'])"`
    tools/detect_unused_modules -m `git --no-pager diff --unified=0 origin/master schedule/* | grep -oP "^-\s+- [\"']?\K.*(?=[\"']?)" | grep -v '{{'`
                                                                        
.PHONY: test-unused-modules                                             
test-unused-modules:                                                    
    tools/detect_unused_modules -a
```
IMHO the script should allow to test locally whatever module user wants to test. Thus I'd limit it with `| grep` in `Makefile`.

Reply:
Well from my understanding there are no test modules outside of `tests/` but please feel free to improve it.

In my case, I deleted existing test module and the CI was failing. That's why I introduced 407cfa0.

Reply:
> But with your change testing `products/*` and `schedule/*` is now not working. Either it should be fixed => we need to revert 407cfa0) or it does not work (not possible to fix it) => we should remove these lines from Makefile.
@pdostal please should we remove these line from `Makefile`as they are filtered out since 407cfa0?
tools/detect_unused_modules -m `git --no-pager diff --unified=0 origin/master products/* | grep -oP "^-.*loadtest.*```
[\"']\K[^\"'].+(?=[\"'])"`
    tools/detect_unused_modules -m `git --no-pager diff --unified=0 origin/master schedule/* | grep -oP "^-\s+- [\"']?\K.*(?=[\"']?)" | grep -v '{{'`
```
Or should we revert 407cfa0 and fix the issue differently?

Reply:
Thx for info. Could you please post the problematic PR / commit?
Also `-a` finds modules with `tests/`, but the others not. That must be fixed.

Reply:
My 407cfa0 commit was follow up to 5cdb327.
I created #13302 as requested in #13246.

So it was because of not removing existing test module but renaming it.

Reply:
Thanks for info. I tested that dd379ea6bbe591f143f971f9e4f6005dfe996c9e with reverted your change 407cfa0d98c2fa5edcae5222c8e9f9cda6fa4b02 is working with my version (and obviously failing with the version from dd379ea6bbe591f143f971f9e4f6005dfe996c9e, which also tries to test string from licence, 
`/home`, `/root`) => going to update commit message and merge.
As the problem was actually also on Tumbleweed, it document that people blindly rely on CI being green (don't run tests locally, don't look at the log output in CI when changing things).